com.calibre_ebook.calibre.yaml

```diff
@@ -21,19 +21,30 @@ finish-args:
 modules:
   - name: calibre
     buildsystem: simple
+    build-options:
+      env:
+        - XDG_UTILS_INSTALL_MODE=system
     build-commands:
-      - mkdir -p /app/bin
-      - tar -x -f calibre*.txz -C /app/bin
+      # Extract Calibre archive to target location
+      - mkdir -p "${FLATPAK_DEST}/lib/calibre"
+      - tar -xvf calibre*.txz -C "${FLATPAK_DEST}/lib/calibre"
       - rm -f calibre*.txz
-      - install -Dm644 /app/bin/resources/images/lt.png /app/share/icons/hicolor/256x256/apps/com.calibre_ebook.calibre.png
-      - install -Dm644 /app/bin/resources/images/tweak.png /app/share/icons/hicolor/256x256/apps/com.calibre_ebook.calibre.ebook_edit.png
-      - install -Dm644 /app/bin/resources/images/viewer.png /app/share/icons/hicolor/256x256/apps/com.calibre_ebook.calibre.ebook_viewer.png
-      - install -Dm644 /app/bin/resources/images/viewer.png /app/share/icons/hicolor/256x256/apps/com.calibre_ebook.calibre.lrfviewer.png
-      - install -Dm644 -t /app/share/metainfo com.calibre_ebook.calibre.appdata.xml
-      - install -Dm644 -t /app/share/applications com.calibre_ebook.calibre.desktop
-      - install -Dm644 -t /app/share/applications com.calibre_ebook.calibre.ebook_edit.desktop
-      - install -Dm644 -t /app/share/applications com.calibre_ebook.calibre.ebook_viewer.desktop
-      - install -Dm644 -t /app/share/applications com.calibre_ebook.calibre.lrfviewer.desktop
+
+      # Ensure that Calibre install script and the `xdg-utils` it uses will not
+      # report errors because they are unable to find any suitable destionation
+      # directory to place their files into
+      - mkdir -p "${FLATPAK_DEST}/bin" "${FLATPAK_DEST}"/share/{applications,desktop-directories,icons/hicolor,mime/packages}
+
+      # Run Calibre system integration scripts
+      - /app/lib/calibre/calibre_postinstall --root="${FLATPAK_DEST}"
+
+      # Rename installed Calibre .desktop files, app icons and appdata XMLs so that Flatpak will not
+      # refuse to use them
+      - python3 ./fixup-exports.py
+    cleanup:
+      - /bin/calibre-uninstall
+      - /share/applications/mimeinfo.cache
+      - /lib/calibre/resources/fonts/liberation  # Included in runtime image
```

Comment:
This will move flatpak aside from upstream a bit (those are in different versions).

@kovidgoyal did you thought about updating liberation fonts in [calibre](https://github.com/kovidgoyal/calibre/tree/master/resources/fonts/liberation)? They seem untouched for many years while [changelog](https://github.com/liberationfonts/liberation-fonts/blob/master/ChangeLog). shows continuous fixes there. You ma download latest version from https://github.com/liberationfonts/liberation-fonts/files/4743886/liberation-fonts-ttf-2.1.1.tar.gz (github is new upstream site for them. As those are binary files I think it can't be submitted to caibre github by anyone else except you.

Reply:
Also what's the actual behaviour when fonts are missing in this path? I see [fedora restores them with symlinks](https://src.fedoraproject.org/rpms/calibre/blob/master/f/calibre.spec#_243).

Reply:
Hmm… I did of course test running Calibre eBook viewer after doing this and could not see anything being different after this. Calibre eBook viewer also lists all fonts installed on my system in the reader settings, so it obviously doesn't ignore the system fonts. (Of course, it be better if Kovid tells us how important they are and if its OK to strip them, that's just my quick observations.)

Reply:
As I recall newer version of Liberation cause BSODs on some windows machines which is why they are not updated in calibre. And if they are not included, something might break, I have no idea, various bits of calibre assume they are always present as a sort of last resort font.

Reply:
Same symlinks for libration on [Debian](https://github.com/norbusan/calibre-debian/blob/4c32dd6209e63746001745095f039c93de3e2b30/debian/rules#L68)

Reply:
So I recommend either making symlinks or dropping that change which isn't related to main purpose of this PR.

Reply:
Dropped for now. To be reevaluated in a different PR.